### DISCUSSION TITLE
A 'NodalVariableIntegralPD' is not a registered object.

### DISCUSSION THREAD

**Discussion by BGBK01** [idaholab/moose#23114] [Q&A General] [answered]
When executing the following Moose postprocessor code, I get the error "A 'NodalVariableIntegralPD' is not a registered object.":

[outlet_surface_integral_left]
  type = NodalVariableIntegralPD
  variable = C_Dm
  boundary = 'SS1'
[]

Is is possible that NodalVariableIntegralPD is not yet fully implemented in Moose?

Any help would be greatly appreciated!
Thanks

**Comment by GiudGiud** [comment, marked as answer]
Hello

which executable/application/module are you using?
Not every object is registered to every application 

guillaume

**Reply by BGBK01** [reply]
So the module is "peridynamics" (moose\modules\peridynamics\src\postprocessors), but my Makefile doesn't seem to have a listing for it: 

![image](https://user-images.githubusercontent.com/122293665/211349931-f144b0d7-bd8b-4089-9a24-603d82734c30.png)

**Reply by GiudGiud** [reply]
you need to add the peridynamics line with yes then
